app/controllers/tasks_controller.rb

```diff
@@ -4,42 +4,72 @@ def index
     render :index
   end
 
+  # Display a specific task
   def show
-    @task = Task.find(params[:id])
+    @task   = Task.find(params[:id])
     render :tasks
   end
 
   def new
     @task = Task.new
-
     render :new
   end
 
-  # Add and save a new task to db
+  # Add and save a new task to list
   def create
-    @task = Task.new(task_params[:task])
-    @task.save
-
+    task = Task.new(task_params[:task])
+    task.save
     index
   end
 
-  # Delete a task from db
+  # Delete a task from list
   def destroy
-    @task = Task.find(params[:id])
-    @task.destroy.save
-
+    task = Task.find(params[:id])
+    task.destroy.save
     redirect_to '/'
   end
 
+  # Update an existing task
+  def edit
+    @task = Task.find(params[:id])
+    render :update
+  end
+
   def update
     @task = Task.find(params[:id])
-    @task.update.save
+    @name = task_params[:task][:name]
+    @desc = task_params[:task][:desc]
+    @comp_date = task_params[:task][:comp_date]
+    @person_id = task_params[:task][:person_id]
```

Comment:
These don't need to be instance variables because they aren't used anywhere except this method.